### ISSUE TITLE
Blacklist certain subdomains from matching URL classes that otherwise allow subdomains to match

### ISSUE THREAD

**Issue by roachcord3** [hydrusnetwork/hydrus#1623]
Fanbox is a good example of this. They allow you to go to `https://username.fanbox.cc` or `https://www.fanbox.cc/@username` to get to the same page. Their API is hosted at `https://api.fanbox.cc`. This means that a URL class for `fanbox.cc` that matches subdomains will match `api.fanbox.cc`, which is not desirable, since that's not the gallery URL for an artist. This can lead to situations where it's hard to define a URL class that contains a redirect to a different subdomain of the same domain, since the class will match its own redirect URL.

So, what I'm requesting is either a subdomain blacklist, or, alternatively, allowing users to use regex in the domain part of the URL class (so that they can define something like `(?!api\.)([a-z0-9][a-z0-9\-]*\.)?fanbox.cc`.

**Comment by floogulinc** [comment]
More specific url classes should already take precedence over less specific ones. It should be the case that a url class with domain specifically `api.fanbox.cc` will be matched before one for `fanbox.cc`.

**Comment by floogulinc** [comment]
If it's not properly prioritizing the one with a defined subdomain I'd consider that a bug then

**Comment by floogulinc** [comment]
I don't think this is an issue but why do you have "should subdomains also match this class" enabled for the `api.fanbox.cc` class?
